hub/core/chunk_engine.py

```diff
@@ -597,36 +603,39 @@ def _samples_to_chunks(
                 samples = samples[num:]
         if register:
             return updated_chunks
+        if progressbar:
+            pbar.close()
         return updated_chunks, tiles
 
-    def _extend(self, samples, update_commit_diff=True):
+    def _extend(self, samples, progressbar, update_commit_diff=True):
         if isinstance(samples, hub.Tensor):
             for sample in samples:
                 self._extend(
-                    [sample], update_commit_diff=update_commit_diff
+                    [sample], progressbar, update_commit_diff=update_commit_diff
```

Comment:
This will show 1 progress bar per sample.

Reply:
This can be simplified as 

```python
if isisnstance(sample, hub.Tensor):
    for sample in tqdm(samples):
      self._extend([sample], progressbar=progressbar, update_commit_diff=update_commit_diff)
```

As to whether it will work, you have to find out yourself.

Reply:
Its exact same thing as how it is now :(